src/operations/command.ts

```diff
@@ -37,7 +37,16 @@ export interface CommandOperationOptions extends OperationOptions, WriteConcernO
 
 /** @internal */
 export interface OperationParent {
-  s: { namespace: MongoDBNamespace };
+  s: {
+    namespace: MongoDBNamespace;
+    serializeFunctions?: boolean;
+    raw?: boolean;
+    promoteLongs?: boolean;
+    promoteValues?: boolean;
+    promoteBuffers?: boolean;
+    ignoreUndefined?: boolean;
+    options?: BSONSerializeOptions;
+  };
   readConcern?: ReadConcern;
```

Comment:
I am not sure if there is a cleaner way to indicate that `s` 1) must have a `namespace` field, 2) may be of type `BSONSerializeOptions`, and 3) may have an `options` field of type `BSONSerializeOptions`. I think it would be nice to avoid listing all of the `BSONSerializeOptions` here.

Reply:
In a couple of places in the codebase (see `AdminPrivate`) we pop out the `s` property to have its own type interface. If you did that here you could then have a `OperationParentPrivate` that `extends` `BsonSerializeOptions`. Then add in the `namespace`. Ideally we dont have the bson options at the top level, as well as  `options`.

Reply:
Assuming we go ahead with moving the Collection bson options into `s.options`, then we would not need any of the bson options directly on the `s` property, which would improve how this block looks. I think we can't avoid having `options` be part of `s` though, because otherwise we can't grab the parent options to do the inheriting.